packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx

```diff
@@ -0,0 +1,378 @@
+  if (pollingTimedOut) {
+    poll = {
+      state: 'error',
+      error: {
+        reasonContents: [
+          <>
+            The command was not run on the server you were trying to add,
+            regenerate command and try again.
+          </>,
+          // TODO (lisa): not sure what this message should be.
+          // <>
+          //   The Teleport Service could not join this Teleport cluster. Check the
+          //   logs for errors by running <br />
+          //   <Mark>kubectl logs -l app=teleport-agent -n {namespace}</Mark>
+          // </>,
+        ],
+      },
+    };
+  } else if (result) {
+    poll = { state: 'success' };
+  }
+
+  return (
+    <Box>
+      <Heading />
+      <Labels
+        labels={props.labels}
+        setLabels={props.setLabels}
+        disableAddBtn={poll.state === 'polling'}
+      />
+      <ButtonSecondary
+        width="200px"
+        disabled={poll.state === 'polling'}
+        onClick={regenerateScriptAndRepoll}
+        mb={3}
+      >
+        Regenerate Command
+      </ButtonSecondary>
+      <CommandWithTimer
+        command={createBashCommand(joinToken.id)}
+        poll={poll}
+        pollingTimeout={timeout}
+      />
+      <ActionButtons
+        onProceed={handleNextStep}
+        disableProceed={poll.state !== 'success' || props.labels.length === 0}
+      />
+    </Box>
+  );
+}
+
+const Heading = () => {
+  return (
+    <>
+      <Header>Deploy a Database Service</Header>
+      <HeaderSubtitle>TODO lorem ipsum dolores</HeaderSubtitle>
+    </>
+  );
+};
+
+export const Labels = ({
+  labels,
+  setLabels,
+  disableAddBtn = false,
+}: {
+  labels: AgentLabel[];
+  setLabels(l: AgentLabel[]): void;
+  disableAddBtn?: boolean;
+}) => {
+  const validator = useValidation() as Validator;
+
+  function addLabel() {
+    // Prevent adding more rows if there are
+    // empty input fields. After checking,
+    // reset the validator so the newly
+    // added empty input boxes are not
+    // considered an error.
+    if (!validator.validate()) {
+      return;
+    }
+    validator.reset();
+    setLabels([...labels, { name: '', value: '' }]);
+  }
+
+  function removeLabel(index: number) {
+    if (labels.length === 1) {
+      // Since at least one label is required
+      // instead of removing the row, clear
+      // the input and turn on error.
+      const newList = [...labels];
+      newList[index].name = '';
+      newList[index].value = '';
+      setLabels(newList);
+
+      validator.validate();
+      return;
+    }
+    const newList = [...labels];
+    newList.splice(index, 1);
+    setLabels(newList);
+  }
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    index: number,
+    labelField: 'key' | 'value'
+  ) => {
+    const { value } = event.target;
+    const newList = [...labels];
+    if (labelField === 'key') {
+      newList[index].name = value;
+    } else {
+      newList[index].value = value;
```

Comment:
[nit]
`labelField` can be `keyof AgentLabel` so we don't need extra layer of mapping: `'key' | 'value'`
```suggestion
    labelField: keyof AgentLabel
  ) => {
    const { value } = event.target;
    const newList = [...labels];
    newList[index] = {...newList[index], [labelField]: value}
```